mne/preprocessing/tests/test_ecg.py

```diff
@@ -55,4 +55,13 @@ def test_find_ecg():
     assert_true(len(w) == 1 and 'unit for channel' in str(w[0].message))
     create_ecg_epochs(raw)
 
+
+def test_ecg_preload():
+    # Test if ECG analysis will work on data that is not preloaded
+    raw = read_raw_fif(raw_fname, preload=False)
+    # There should be no ECG channels, or else preloading will not be
+    # tested
+    assert_true('ecg' not in raw)
+    create_ecg_epochs(raw)
```

Comment:
I updated the existing test to check the new functionality by not preloading the data. Let me know if that works